load_balancers.go

```diff
@@ -103,6 +104,11 @@ func (l LoadBalancer) AsRequest() *LoadBalancerRequest {
 		r.Region = l.Region.Slug
 	}
 
+	if l.Firewall != nil {
+		r.Firewall = &LBFirewall{}
+		*r.Firewall = *l.Firewall
```

Comment:
There's some subtlety that needs to be taken into account here. The pointer dereference copy works for the other types b/c they only have primitives. B/c firewalls has lists, it's still copying the pointers to the list. We'll need to do something like `append([]int(nil), l.DropletIDs...),` for `DropletIDs` above to copy the elements over for both the allow and deny lists.

Reply:
good point; i think we should just make a `deepCopy` helper?

Reply:
https://github.com/digitalocean/godo/pull/585/files#diff-c9b4c15afcc0c5f0ead492bd8e12abb791b72cdd4985919a15f04b7708629281R165

🙏